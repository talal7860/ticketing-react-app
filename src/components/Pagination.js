import React from 'react';
import { Pagination as BootstrapPagination, PaginationItem, PaginationLink } from 'reactstrap';
import PropTypes from 'prop-types';

const Pagination = ({ onPageChange, meta  }) => {
	let previousDisabled = true;
	let nextDisabled = true;
	if (meta) {
		if (meta.total_pages === 1) {
			nextDisabled = true;
			previousDisabled = true;
		} else if (meta.total_pages === meta.current_page) {
			nextDisabled = true;
			previousDisabled = false;
		} else if (meta.current_page === 1) {
			nextDisabled = false;
			previousDisabled = true;
		} else {
			nextDisabled = false;
			previousDisabled = false;
		}
	}
	return (
		<BootstrapPagination className='float-right'>
			<PaginationItem disabled={previousDisabled}>
				<PaginationLink previous href="#" onClick={(e) => onPageChange(e, meta.current_page - 1)} />
			</PaginationItem>
			<PaginationItem disabled={nextDisabled}>
				<PaginationLink next href="#" onClick={(e) => onPageChange(e, meta.current_page + 1)} />
			</PaginationItem>
		</BootstrapPagination>
	);
};

Pagination.propTypes = {
  onPageChange: PropTypes.func.isRequired,
  meta: PropTypes.object,
};

Pagination.defaultProps = {
  meta: {},
};

export default Pagination;
