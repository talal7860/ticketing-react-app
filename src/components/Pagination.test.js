import React from 'react';
import { shallow } from 'enzyme';
import Pagination from './Pagination';

describe('Pagination', () => {
  it('should render a Pagination Component successfully', () => {
    const wrapper = shallow(<Pagination onPageChange={() => {}} />);
    expect(wrapper).toMatchSnapshot();
  });

});

