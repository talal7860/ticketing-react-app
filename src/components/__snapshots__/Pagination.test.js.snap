// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination should render a Pagination Component successfully 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Pagination
    meta={Object {}}
    onPageChange={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <PaginationItem
          disabled={false}
          tag="li"
>
          <PaginationLink
                    href="#"
                    onClick={[Function]}
                    previous={true}
                    tag="a"
          />
</PaginationItem>,
        <PaginationItem
          disabled={true}
          tag="li"
>
          <PaginationLink
                    href="#"
                    next={true}
                    onClick={[Function]}
                    tag="a"
          />
</PaginationItem>,
      ],
      "className": "float-right",
      "tag": "ul",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <PaginationLink
            href="#"
            onClick={[Function]}
            previous={true}
            tag="a"
/>,
          "disabled": false,
          "tag": "li",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "href": "#",
            "onClick": [Function],
            "previous": true,
            "tag": "a",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <PaginationLink
            href="#"
            next={true}
            onClick={[Function]}
            tag="a"
/>,
          "disabled": true,
          "tag": "li",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "href": "#",
            "next": true,
            "onClick": [Function],
            "tag": "a",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <PaginationItem
            disabled={false}
            tag="li"
>
            <PaginationLink
                        href="#"
                        onClick={[Function]}
                        previous={true}
                        tag="a"
            />
</PaginationItem>,
          <PaginationItem
            disabled={true}
            tag="li"
>
            <PaginationLink
                        href="#"
                        next={true}
                        onClick={[Function]}
                        tag="a"
            />
</PaginationItem>,
        ],
        "className": "float-right",
        "tag": "ul",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <PaginationLink
              href="#"
              onClick={[Function]}
              previous={true}
              tag="a"
/>,
            "disabled": false,
            "tag": "li",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "href": "#",
              "onClick": [Function],
              "previous": true,
              "tag": "a",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <PaginationLink
              href="#"
              next={true}
              onClick={[Function]}
              tag="a"
/>,
            "disabled": true,
            "tag": "li",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "href": "#",
              "next": true,
              "onClick": [Function],
              "tag": "a",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
